File: BOJ/silver4/boj_10845.py
# 10845. 큐

import sys
input = sys.stdin.readline

n = int(input())
q = []
for _ in range(n):
    command = input().split()
    if command[0] == 'push':
        q.append(command[1])
    elif command[0] == 'pop':
        if not q:
            print(-1)
        else:
            print(q.pop(0))
    elif command[0] == 'size':
        print(len(q))
    elif command[0] == 'empty':
        if not q:
            print(1)
        else:
            print(0)
    elif command[0] == 'front':
        if not q:
            print(-1)
        else:
            print(q[0])
    elif command[0] == 'back':
        if not q:
            print(-1)
        else:
            print(q[-1])

File: BOJ/silver4/boj_10845.cpp
// 10845. 큐
#include <bits/stdc++.h>
using namespace std;
int main(void)
{
  ios::sync_with_stdio(0);
  cin.tie(0);
  queue<int> q;
  int n;
  cin >> n;
  string s;
  int t;
  while (n--)
  {
    cin >> s;
    if (s == "push")
    {
      cin >> t;
      q.push(t);
    }
    else if (s == "pop")
    {
      if (!q.empty())
      {
        cout << q.front() << "\n";
        q.pop();
      }
      else
        cout << -1 << "\n";
    }
    else if (s == "size")
    {
      cout << q.size() << "\n";
    }
    else if (s == "empty")
    {
      cout << q.empty() << "\n";
    }
    else if (s == "front")
    {
      if (!q.empty())
        cout << q.front() << "\n";
      else
        cout << -1 << "\n";
    }
    else
    {
      if (!q.empty())
        cout << q.back() << "\n";
      else
        cout << -1 << "\n";
    }
  }
}